// react-native-month/lib/commonjs/utils/validations.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.changeMarkedDays = changeMarkedDays;
exports.changedDate = changedDate;
exports.isDay = isDay;
exports.isLeapYear = isLeapYear;
exports.isMonth = isMonth;
exports.isYear = isYear;
exports.leapYear = leapYear;
exports.sameDate = sameDate;
const MAX_DATE = new Date(8640000000000000);
const MIN_DATE = new Date(-8640000000000000);

function isDay(day) {
  return Number.isInteger(day) && day >= 1 && day <= 31;
}

function isMonth(month) {
  return Number.isInteger(month) && month >= 0 && month < 12;
}

function isYear(year) {
  return Number.isInteger(year) && year >= MIN_DATE.getFullYear() && year <= MAX_DATE.getFullYear();
}

function leapYear(year) {
  return year % 4 === 0 && year % 100 !== 0 || year % 400 === 0;
}

function isLeapYear(year) {
  return isYear(year) && leapYear(year);
}

function sameDate(one, other) {
  return one.getDate() === other.getDate() && one.getMonth() === other.getMonth() && one.getFullYear() === other.getFullYear();
}

function changedDate(one, other) {
  return one instanceof Date !== other instanceof Date || one instanceof Date && other instanceof Date && !sameDate(one, other);
}

function changeMarkedDays() {
  let markedDays = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  let nextMarkedDays = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
  return Object.keys(markedDays).length !== Object.keys(nextMarkedDays).length;
}
